File: easy/Details/Details.cs
using System;
using System.IO;
using System.Collections.Generic;

class Program
{
    static void Main(string[] args)
    {
        using (StreamReader reader = File.OpenText(args[0]))
        while (!reader.EndOfStream)
        {
            string line = reader.ReadLine();
            if (null == line)
                continue;
            // do something with line
            ShowMoveSteps(line);
        }
    }
    
    static void ShowMoveSteps(string line){
            string[] words = line.Split(',');
            int row = words.Length;
            int col = words[0].Length;
            int steps = col;
            for(int i=0;i<row;i++){
                int intSteps = 0;
                bool inX = true;
                bool inY = false;
                int j = 0;
                while(!inY){
                    inX = words[i][j].Equals('X'); 
                    inY = words[i][j].Equals('Y'); 
                    if(inX) intSteps =0;
                    if(!inX && !inY)intSteps++;
                    j++;
                }
                if (intSteps<steps)steps = intSteps;
            }
            Console.WriteLine(steps);
        }
}

File: easy/Mixed-Content/Details.java
import java.io.*;
public class Main {
    public static void main (String[] args) throws IOException {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            line = line.trim();
            ShowMoveSteps(line);
        }
    }
    
    static void ShowMoveSteps(String line){
        String[] words = line.split(",");
        int row = words.length;
        int col = words[0].length();
        int steps = col;
        for(int i=0;i<row;i++){
            int intSteps = 0;
            boolean inX = true;
            boolean inY = false;
            int j = 0;
            while(!inY){
                inX = words[i].charAt(j) == 'X'; 
                inY = words[i].charAt(j) == 'Y'; 
                if(inX) intSteps =0;
                if(!inX && !inY)intSteps++;
                j++;
            }
            if (intSteps<steps)steps = intSteps;
        }
        System.out.println(steps);
    }
}
